main.ts

```diff
@@ -84,6 +93,136 @@ if ($searchInput) {
     });
   });
 }
+
+// Starting from here is mostly issue 2
+
+// function use to show information of individual card.
+
+function showCharacterDetail(charData: Data): void {
+  const $characterPic = document.querySelector(
+    '#character-image',
+  ) as HTMLImageElement;
+  const $characterFilm = document.querySelector(
+    '.characterFilm',
+  ) as HTMLElement;
+  const $characterVideogames = document.querySelector(
+    '.characterVideogames',
+  ) as HTMLElement;
+  const $characterInfo = document.querySelector(
+    '.characterInfo',
+  ) as HTMLElement;
+
+  $characterPic.src = charData.imageUrl;
+  $characterPic.alt = charData.name;
+  $characterFilm.textContent = charData.films.join(' , ');
+  $characterVideogames.textContent = charData.videoGames.join(' ');
+  $characterInfo.innerHTML = `<a href="${charData.sourceUrl}">${charData.sourceUrl}</a>`;
+
+  // main page swapping
+  const $mainPage = document.querySelector(
+    '[data-view="main-page"]',
+  ) as HTMLElement;
+  const $characterDetail = document.querySelector(
+    '[data-view="character-page"]',
+  ) as HTMLElement;
```

Comment:
Move these to the global space. You want to try and do your querySelectors globally so you only need to search the DOM once. You also use these multiple times so rather than needing to create these variables multiple times, we can just create them once!